src/pydvl/utils/dataset.py

```diff
@@ -155,15 +159,11 @@ def get_training_data(self, train_indices: Optional[Iterable[int]]):
             y = self.y_train[train_indices]
             return x, y
 
-    def get_test_data(self, test_indices: Optional[Iterable[int]]):
-        """Given a set of indices, returns the test data that refer to those
-        indices."""
-        if test_indices is None:
-            return self.x_test, self.y_test
-        else:
-            x = self.x_test[test_indices]
-            y = self.y_test[test_indices]
-            return x, y
+    def get_test_data(
```

Comment:
I kept the argument because I wanted to keep the interface the same as for `get_training_data()`.

For data valuation we always return the entire test set, so we can safely ignore the indices. But for feature valuation we need in the indices and for that I can just overwrite the `get_test_data()` method in a subclass and the Utility remains the same.

Reply:
I'm not sure I get the need for feature valuation, so I'll let you be the judge of this. However, you have the same behaviour if you leave the possibility of returning just a few indices or all of them if the arg is `None`.

Reply:
See my comment below.